c/tskit/trees.h

```diff
@@ -362,42 +363,42 @@ int tsk_treeseq_f4(tsk_treeseq_t *self, tsk_size_t num_sample_sets,
 int tsk_tree_init(tsk_tree_t *self, tsk_treeseq_t *tree_sequence, tsk_flags_t options);
 int tsk_tree_free(tsk_tree_t *self);
 
-tsk_id_t tsk_tree_get_index(tsk_tree_t *self);
-tsk_size_t tsk_tree_get_num_roots(tsk_tree_t *self);
+tsk_id_t tsk_tree_get_index(const tsk_tree_t *self);
+tsk_size_t tsk_tree_get_num_roots(const tsk_tree_t *self);
 
 int tsk_tree_first(tsk_tree_t *self);
 int tsk_tree_last(tsk_tree_t *self);
 int tsk_tree_next(tsk_tree_t *self);
 int tsk_tree_prev(tsk_tree_t *self);
 int tsk_tree_clear(tsk_tree_t *self);
 
-void tsk_tree_print_state(tsk_tree_t *self, FILE *out);
+void tsk_tree_print_state(const tsk_tree_t *self, FILE *out);
 /** @} */
 
 int tsk_tree_set_root_threshold(tsk_tree_t *self, tsk_size_t root_threshold);
-tsk_size_t tsk_tree_get_root_threshold(tsk_tree_t *self);
+tsk_size_t tsk_tree_get_root_threshold(const tsk_tree_t *self);
 
-bool tsk_tree_has_sample_lists(tsk_tree_t *self);
-bool tsk_tree_has_sample_counts(tsk_tree_t *self);
-bool tsk_tree_equals(tsk_tree_t *self, tsk_tree_t *other);
+bool tsk_tree_has_sample_lists(const tsk_tree_t *self);
+bool tsk_tree_has_sample_counts(const tsk_tree_t *self);
+bool tsk_tree_equals(const tsk_tree_t *self, const tsk_tree_t *other);
 /* Returns true if u is a descendant of v; false otherwise */
-bool tsk_tree_is_descendant(tsk_tree_t *self, tsk_id_t u, tsk_id_t v);
-bool tsk_tree_is_sample(tsk_tree_t *self, tsk_id_t u);
+bool tsk_tree_is_descendant(const tsk_tree_t *self, tsk_id_t u, tsk_id_t v);
+bool tsk_tree_is_sample(const tsk_tree_t *self, tsk_id_t u);
 
 int tsk_tree_copy(tsk_tree_t *self, tsk_tree_t *dest, tsk_flags_t options);
 int tsk_tree_set_tracked_samples(
     tsk_tree_t *self, size_t num_tracked_samples, tsk_id_t *tracked_samples);
 int tsk_tree_set_tracked_samples_from_sample_list(
     tsk_tree_t *self, tsk_tree_t *other, tsk_id_t node);
 
-int tsk_tree_get_parent(tsk_tree_t *self, tsk_id_t u, tsk_id_t *parent);
-int tsk_tree_get_time(tsk_tree_t *self, tsk_id_t u, double *t);
-int tsk_tree_get_mrca(tsk_tree_t *self, tsk_id_t u, tsk_id_t v, tsk_id_t *mrca);
-int tsk_tree_get_num_samples(tsk_tree_t *self, tsk_id_t u, size_t *num_samples);
+int tsk_tree_get_parent(const tsk_tree_t *self, tsk_id_t u, tsk_id_t *parent);
+int tsk_tree_get_time(const tsk_tree_t *self, tsk_id_t u, double *t);
+int tsk_tree_get_mrca(const tsk_tree_t *self, tsk_id_t u, tsk_id_t v, tsk_id_t *mrca);
+int tsk_tree_get_num_samples(const tsk_tree_t *self, tsk_id_t u, size_t *num_samples);
 int tsk_tree_get_num_tracked_samples(
-    tsk_tree_t *self, tsk_id_t u, size_t *num_tracked_samples);
+    const tsk_tree_t *self, tsk_id_t u, size_t *num_tracked_samples);
 int tsk_tree_get_sites(tsk_tree_t *self, tsk_site_t **sites, tsk_size_t *sites_length);
```

Comment:
const self here?

Reply:
I've got `tsk_tree_get_sites` flipped to const in the final branch/PR `pr/const_c_api`. I've been organizing this change as part of the "changes in tsk_tree_t type" since they collectively trigger way more changes in other parts of the codebase.